File: ungs-docsys-ms/src/main/java/com/ungs/docsys/controllers/AppUserController.java
package com.ungs.docsys.controllers;

import com.ungs.docsys.dtos.AppUserRequestDto;
import com.ungs.docsys.dtos.AppUserSignInResponseDto;
import com.ungs.docsys.dtos.AppUserResponseDto;
import com.ungs.docsys.dtos.AppUserAdminResponseDto;
import com.ungs.docsys.services.AppUserService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import jakarta.validation.Valid;
import lombok.AllArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;
import org.springframework.security.access.prepost.PreAuthorize;

import java.util.List;

@RestController
@RequestMapping("/v1/users")
@AllArgsConstructor
@Validated
public class AppUserController {

    private final AppUserService appUserService;

    @Operation(summary = "Sign In")
    @ApiResponse(responseCode = "201", description = "Successfully signed in")
    @ApiResponse(responseCode = "400", description = "Invalid input data")
    @PostMapping("/signIn")
    public ResponseEntity<AppUserSignInResponseDto> signIn(@Valid @RequestBody AppUserRequestDto request) {
        AppUserSignInResponseDto appUserSignInResponseDto = appUserService.singIn(request);
        return ResponseEntity.status(HttpStatus.OK).body(appUserSignInResponseDto);
    }

    @GetMapping
    @PreAuthorize("hasRole('ADMIN')")
    @Operation(summary = "Get all users (admin only)")
    @ApiResponse(responseCode = "200", description = "List of users retrieved successfully")
    public ResponseEntity<List<AppUserAdminResponseDto>> getAllUsers() {
        return ResponseEntity.ok(appUserService.getAllUsersForAdmin());
    }
}

File: ungs-docsys-bff/src/modules/login/app-user/controllers/app-user.controller.ts
import { Controller, Post, Body } from "@nestjs/common";
import { AppUserService } from "../services/app-user.service";
import { AppUserRequestDto } from "../dtos/app-user-request.dto";
import { AppUserResponseDto } from "../dtos/app-user-response.dto";
import { ApiOperation, ApiResponse } from "@nestjs/swagger";

@Controller('/v1/users/signIn')
export class AppUserController {
  constructor(private readonly appUserService: AppUserService) {}

  @Post()
  @ApiOperation({ summary: 'Sign In' })
  @ApiResponse({
    status: 200,
    description: 'Successfully signed in',
    type: AppUserResponseDto,
  })
  async signIn(@Body() request: AppUserRequestDto): Promise<AppUserResponseDto> {
    return await this.appUserService.signIn(request);
  }
}
